ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -388,6 +360,12 @@ func (r *RayJobReconciler) createNewK8sJob(ctx context.Context, rayJobInstance *
 		},
 	}
 
+	// Without TTLSecondsAfterFinished, the job has a default deletion policy of `orphanDependents` causing
```

Comment:
https://kubernetes.io/docs/concepts/workloads/controllers/job/#ttl-mechanism-for-finished-jobs